### ISSUE TITLE
Setting a page as child of root, throws error but still updates structure tree

### ISSUE THREAD

**Issue by johncarter-** [statamic/cms#2763]
1. Clean install 3.0.21
2. Create new entry in Pages collection
3. Set the entry's parent to the Home page.
4. Get an error thrown as notification.
5. But the collection tree still updated, despite the error.
6. Then because the tree is saved but not valid, the collection listing view is spannered. As per screenshot below

Might be related to #2647

![image](https://user-images.githubusercontent.com/3776888/97551441-e4581c80-19ca-11eb-9b6a-336afe01266e.png)

**Statamic version:** 3.0.21
**PHP version:** 7.4
**`Fresh install from `statamic/statamic`**

**Comment by edalzell** [comment]
we hit this as well, If you take an existing nested page, and set it's `parent` to the home page, it breaks the site.

**Comment by jelleroorda** [comment]
Same here, temporarily worked around it by adding the follow snippet to the `parent` field of the pages blueprint:

```yaml
      -
        handle: parent
        field:
          type: entries
          collections:
            - pages
          max_items: 1
          listable: false
          localizable: true
          mode: default
          unless:
            slug: 'equals home'

```

**Comment by jelleroorda** [comment]
I think this issue should have been fixed through #3104
